File: check_channels.py
import pymongo

try:
    # Подключение к MongoDB
    client = pymongo.MongoClient('mongodb://127.0.0.1:27017/')
    db = client['rocketchat']

    # Получаем все каналы (тип 'c')
    rooms = db.rocketchat_room.find({'t': 'c'}, {'name': 1, 'fname': 1})

    print('=== КАНАЛЫ ROCKET.CHAT ===')
    for room in rooms:
        name = room.get('name', 'N/A')
        fname = room.get('fname', 'N/A')
        print(f'- name: "{name}", fname: "{fname}"')

        # Проверим кодировку fname
        if fname != 'N/A':
            try:
                fname_encoded = fname.encode('utf-8')
                print(f'  UTF-8 bytes: {fname_encoded}')
            except:
                print(f'  ОШИБКА КОДИРОВКИ fname')

    client.close()
    print('\n=== ПРОВЕРКА ЗАВЕРШЕНА ===')

except Exception as e:
    print(f'ОШИБКА: {e}')

File: check_channels.js
// Проверка каналов в Rocket.Chat
use rocketchat;

print("=== КАНАЛЫ В БАЗЕ ===");
db.rocketchat_room.find({t: "c"}, {name: 1, _id: 1}).forEach(function(doc) {
    print("Канал: " + doc.name + " | ID: " + doc._id);
});

print("\n=== ПОДПИСКИ ПОЛЬЗОВАТЕЛЯ owner ===");
db.rocketchat_subscription.find({u: {$regex: "owner"}}, {name: 1, rid: 1, roles: 1}).forEach(function(doc) {
    print("Подписка: " + doc.name + " | RoomID: " + doc.rid + " | Roles: " + JSON.stringify(doc.roles));
});

print("\n=== НАСТРОЙКИ SETUP WIZARD ===");
db.rocketchat_settings.find({_id: "Show_Setup_Wizard"}, {value: 1}).forEach(function(doc) {
    print("Setup Wizard: " + doc.value);
});
